experimental/HttpBenchmark/Program.cs

```diff
@@ -0,0 +1,375 @@
+            requestBytes = Encoding.ASCII.GetBytes(requestString);
+            responseLength *= pipeline;
+
+            var threadObjects = new Thread[connections];
+            for (var i = 0; i < connections; i++)
+            {
+                var thread = new Thread(() =>
+                {
+                    var responseBuffer = new byte[responseLength];
+
+                    if (keepalive)
+                    {
+                        using (var socket = CreateSocket(uri).Result)
+                        {
+                            Interlocked.Increment(ref _connections);
+
+                            while (true)
+                            {
+                                SendReceive(socket, requestBytes, responseLength, responseBuffer);
+                                Interlocked.Add(ref _requests, pipeline);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        while (true)
+                        {
+                            using (var socket = CreateSocket(uri).Result)
+                            {
+                                Interlocked.Increment(ref _connections);
+                                SendReceive(socket, requestBytes, responseLength, responseBuffer);
+                                Interlocked.Add(ref _requests, pipeline);
+                            }
+                            Interlocked.Decrement(ref _connections);
+                        }
+                    }
+                });
+                threadObjects[i] = thread;
+            }
+
+            _stopwatch.Start();
+
+            for (var i = 0; i < connections; i++)
+            {
+                threadObjects[i].Start();
+            }
+
+            for (var i = 0; i < connections; i++)
+            {
+                threadObjects[i].Join();
+            }
+
+            return 0;
+        }
+
+        private static bool ResponseComplete(byte[] buffer, int count)
+        {
+            var response = Encoding.ASCII.GetString(buffer, 0, count);
+
+            // Determine if response uses Content-Length or Chunked
+            int? contentLength = null;
+            bool chunked = false;
+            using (var reader = new StringReader(response))
+            {
+                string line;
+                while ((line = reader.ReadLine()) != null)
+                {
+                    if (line.StartsWith("Content-Length: "))
+                    {
+                        contentLength = int.Parse(line.Substring("Content-Length: ".Length));
+                        break;
+                    }
+                    else if (line == "Transfer-Encoding: chunked")
+                    {
+                        chunked = true;
+                        break;
+                    }
+                }
+            }
+
+            if (contentLength != null)
```

Comment:
@rynowak
rynowak Pending
Would it be possible to also check the status code here? I have the heebie-jeebies about any sort of automation that can report perf information when the run wasn't successful.